### PR TITLE
feat: List of CIS docker benchmark results

### PR DESCRIPTION
**Pull request by pbalogh-sa**
Signed-off-by: Peter Balogh <p.balogh.sa@gmail.com>

### CODE REVIEW

File: api/swagger.yaml [hunk 1]
```diff
@@ -1436,6 +1444,32 @@ paths:
         default:
           $ref: '#/responses/UnknownError'
 
+  /cisdockerbenchmarkresults/{id}:
+    get:
+      summary: 'Get CIS Docker Benchmark results by Application Resource'
```
Comment (akpsgit) [inline api/swagger.yaml:1449]: Get CIS Docker Benchmark results by Application Resource ID

File: api/swagger.yaml [hunk 2]
```diff
@@ -1436,6 +1444,32 @@ paths:
...
+            type: 'object'
+            required:
+              - total
+            properties:
+              total:
+                type: 'integer'
+                description: 'Total CIS docker benchmark results by Application Resource'
+              items:
+                type: 'array'
+                description: 'List of CIS docker benchmark results by Application Resource'
```
Comment (akpsgit) [inline api/swagger.yaml:1467]: List of CIS docker benchmark findings for an Application Resource

File: api/swagger.yaml [hunk 3]
```diff
@@ -1436,6 +1444,32 @@ paths:
...
+      summary: 'Get CIS Docker Benchmark results by Application Resource'
+      parameters:
+        - $ref: '#/parameters/id'
+        - $ref: '#/parameters/cisDockerBenchmarkResultsSortKey'
+        - $ref: '#/parameters/sortDir'
+      responses:
+        '200':
+          description: 'Success'
+          schema:
+            type: 'object'
```
Comment (akpsgit) [inline api/swagger.yaml:1458]: maybe it's better to define a new object for the CIS docker benchmark result (that includes the total and the list of findings)?

Comment (pbalogh-sa) [reply api/swagger.yaml:1458]: In the case of the other API endpoints where you can sort the result, the total filed is separated. So if we want to be consequent we should do the same here. WDYT?

File: api/swagger.yaml (hunk 2 above)
Comment (pbalogh-sa) [reply api/swagger.yaml:1467]: ack

File: api/swagger.yaml (hunk 1 above)
Comment (pbalogh-sa) [reply api/swagger.yaml:1449]: ack

File: backend/pkg/database/cis_docker_benchmark_result.go [hunk 4]
```diff
@@ -16,32 +16,48 @@
...
+}
+
 type CISDockerBenchmarkResultTable interface {
 	CountPerLevel(filters *CountFilters) ([]*models.CISDockerBenchmarkLevelCount, error)
+	GetCISDockerBenchmarkResultsAndTotal(params operations.GetCisdockerbenchmarkresultsIDParams) ([]CISDockerBenchmarkCheckView, int64, error)
 }
 
 type CISDockerBenchmarkResultTableHandler struct {
-	table *gorm.DB
+	applicationsCisDockerBenchmarkChecksWiev *gorm.DB
```
Comment (akpsgit) [inline backend/pkg/database/cis_docker_benchmark_result.go:59]: Wiev

File: backend/pkg/database/cis_docker_benchmark_result.go [hunk 5]
```diff
@@ -75,3 +91,58 @@ func (c *CISDockerBenchmarkResultTableHandler) setCountFilters(tx *gorm.DB, filt
...
+		return nil, 0, err
+	}
+
+	return cisDockerBenchmarkResults, count, nil
+}
+
+func CISDockerBenchmarkResultFromDB(result *CISDockerBenchmarkCheckView) *models.CISDockerBenchmarkAssessment {
+	return &models.CISDockerBenchmarkAssessment{
+		Code:  result.Code,
+		Title: result.Titles,
```
Comment (akpsgit) [inline backend/pkg/database/cis_docker_benchmark_result.go:124]: why is it called titles and descriptions (in plural)?

File: backend/pkg/rest/cisdockerbenchmark_controller.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+		return operations.NewGetCisdockerbenchmarkresultsIDDefault(http.StatusInternalServerError).
+			WithPayload(
+				&models.APIResponse{
+					Message: "Oops",
+				},
+			)
+	}
+
+	log.Debugf("GetCISDockerBenchmarkResults controller was invoked. "+
+		"params=%+v, vulnerabilitiesView=%+v, total=%+v", params, cisDockerBenchmarkResults, total)
```
Comment (akpsgit) [inline backend/pkg/rest/cisdockerbenchmark_controller.go:42]: vulnerabilitiesView->cisDockerBenchmarkResults

File: backend/pkg/types/cis_docker_benchmark.go [hunk 7]
```diff
@@ -40,6 +41,7 @@ func cisDockerBenchmarkResultFromBackendAPI(in *models.CISDockerBenchmarkCodeInf
 	return &CISDockerBenchmarkResult{
 		Code:         in.Code,
 		Level:        in.Level,
+		Titles:       getTitlesFromBackendAPI(in.Assessments),
```
Comment (akpsgit) [inline backend/pkg/types/cis_docker_benchmark.go:44]: not sure i understand. a specific code can have multiple titles and descriptions?

File: backend/pkg/database/cis_docker_benchmark_result.go (hunk 4 above)
Comment (pbalogh-sa) [reply backend/pkg/database/cis_docker_benchmark_result.go:59]: thx

File: backend/pkg/database/cis_docker_benchmark_result.go [hunk 8]
```diff
@@ -16,32 +16,48 @@
...
+
 type CISDockerBenchmarkResultTable interface {
 	CountPerLevel(filters *CountFilters) ([]*models.CISDockerBenchmarkLevelCount, error)
+	GetCISDockerBenchmarkResultsAndTotal(params operations.GetCisdockerbenchmarkresultsIDParams) ([]CISDockerBenchmarkCheckView, int64, error)
 }
 
 type CISDockerBenchmarkResultTableHandler struct {
-	table *gorm.DB
+	applicationsCisDockerBenchmarkChecksWiev *gorm.DB
+	cisDockerBenchmarkChecksWiev             *gorm.DB
```
[the remark this answers is not in the document]
Comment (pbalogh-sa) [reply backend/pkg/database/cis_docker_benchmark_result.go:60]: thx

File: backend/pkg/rest/cisdockerbenchmark_controller.go (hunk 6 above)
Comment (pbalogh-sa) [reply backend/pkg/rest/cisdockerbenchmark_controller.go:42]: thx

File: backend/pkg/types/cis_docker_benchmark.go (hunk 7 above)
Comment (pbalogh-sa) [reply backend/pkg/types/cis_docker_benchmark.go:44]: Maybe it's better to get `title` based on the code.

File: backend/pkg/database/database.go [hunk 9]
```diff
@@ -314,7 +315,18 @@ GROUP BY vulnerabilities.id,
          p.name,
          p.version,
          p.id,
-         pv.fix_version
+         pv.fix_version;
+  `
+
+	cisDockerBenchmarkResultsViewQuery = `
+CREATE VIEW cis_d_b_checks_view AS
+SELECT resource_cis_d_b_checks.*,
```
Comment (FrimIdan) [inline backend/pkg/database/database.go:323]: do you need all `resource_cis_d_b_checks` fields or only resource id?

Comment (pbalogh-sa) [reply backend/pkg/database/database.go:323]: No, you are right. Thx

File: backend/pkg/database/database.go [hunk 10]
```diff
@@ -314,7 +315,18 @@ GROUP BY vulnerabilities.id,
          p.name,
          p.version,
          p.id,
-         pv.fix_version
+         pv.fix_version;
+  `
+
+	cisDockerBenchmarkResultsViewQuery = `
+CREATE VIEW cis_d_b_checks_view AS
+SELECT resource_cis_d_b_checks.resource_id,
```
Comment (FrimIdan) [inline backend/pkg/database/database.go:323]: `resource_cis_d_b_checks.resource_id AS resource_id`

Comment (pbalogh-sa) [reply backend/pkg/database/database.go:323]: It will be resource_id if the `AS` is not defined. But I agree with you because it is more evident. THX

**Review by akpsgit — APPROVED** [review]

**Pull request merged**